lib/config/recommended-gts.js

```diff
@@ -1,4 +1,17 @@
 const base = require('./base');
 const gtsRules = require('../recommended-rules-gts');
 
-module.exports = [...base, { rules: gtsRules }];
+let typescriptRecommendedRules = {};
+try {
+  // typescript recommended rules turn off eslint rules that do not work for ts/gts
+  // but they only to that for known extensions, therefore we need to reapply them in our
+  // recommended config
+  // see issue https://github.com/typescript-eslint/typescript-eslint/issues/8607
+  const recommended = // eslint-disable-next-line n/no-extraneous-require
+    require('@typescript-eslint/eslint-plugin').configs['eslint-recommended'];
+  typescriptRecommendedRules = recommended.overrides[0].rules;
```

Comment:
Is this really safe? It assumes the structure of a config in a third-party plugin which presumably is subject to change and could differ across versions.

Reply:
I think it's safe because it has to follow the structure eslint needs.
It might happen someday that they will not use override and file filter. In that case it just fails and we keep the empty rules object. 
In that case the eslint ts config could be used normally

Reply:
what if typescript-eslint puts the config at a different overrides position though?

is there a way to tell typescript-eslint to check extra files?
surely, svelte / vue / etc have had to do something similar?